relay-server/src/services/store.rs

```diff
@@ -968,6 +968,7 @@ impl StoreService {
 
         let is_segment = span.is_segment;
         let has_parent = span.parent_span_id.is_some();
+        let platform = VALID_PLATFORMS.iter().find(|p| *p == &span.platform);
```

Comment:
Checking `VALID_PLATFORMS` to make sure cardinality is low, and to get a static `&str`.